src/aspire/classification/averager2d.py

```diff
@@ -336,17 +337,33 @@ def align(self, classes, reflections, basis_coefficients):
 
                 # then store dot between class base image (0) and each nbor
                 for j, nbor in enumerate(rotated_nbrs):
-                    results[j, i] = np.dot(nbr_coef[0], nbor)
+                    _correlations[j, i] = np.dot(nbr_coef[0], nbor)
 
             # Now along each class, find the index of the angle reporting highest correlation
-            angle_idx = np.argmax(results, axis=1)
+            angle_idx = np.argmax(_correlations, axis=1)
 
-            # Store that angle as our rotation for this image
-            rotations[k, :] = test_angles[angle_idx]
+            # Take the correlation corresponding to angle_idx
+            _correlations = np.take_along_axis(
+                _correlations, np.expand_dims(angle_idx, axis=1), axis=1
+            ).reshape(n_nbor)
 
-            # Also store the correlations for each neighbor
-            for j in range(n_nbor):
-                correlations[k, j] = results[j, angle_idx[j]]
+            return test_angles[angle_idx], _correlations
+
+        if self.num_procs <= 1:
```

Comment:
Isn't `self.num_procs` a positive integer? Should this be `if self.num_procs == 1:`

Reply:
That's true, but I'd like to preserve the `0` or `-1` cases in the future if we need to differentiate `1`.

`1` could potentially be run in either mode, perhaps for debugging one day.  You are totally correct that the base class is currently `>0` and that we currently do not want to stand up multiprocessing for a count of 1.

I think I might leave these for someone in the future self though...